be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/DataBuilder.scala

```diff
@@ -33,11 +35,28 @@ object DataBuilder {
     case _ => throw new ProtocolException(s"Unknown object '${_object}' encountered while creating a Data")
   }
 
-  private def buildLaoData(action: ActionType, payload: String): MessageData = action match {
-    case ActionType.CREATE => CreateLao.buildFromJson(payload)
-    case ActionType.STATE => StateLao.buildFromJson(payload)
-    case ActionType.UPDATE_PROPERTIES => UpdateLao.buildFromJson(payload)
-    case _ => throw new ProtocolException(s"Unknown action '$action' encountered while creating a Lao Data")
+  /**
+    * builds MessageData from payload conformly to the protocol
+    * after validating its schema
+    * @param action action type
+    * @param payload message data in json string format
+    * @return parsed MessageData if the validation and build succedds, throws ProtocolException otherwise
+    */
+  @throws(classOf[ProtocolException])
+  private def buildLaoData(action: ActionType, payload: String): MessageData = {
+    val laoMetadataMap: Map[ActionType, MetaData] = DataRegistryModule.REGISTRY.getFromObject(ObjectType.LAO)
+    action match {
+      case ActionType.CREATE =>
+        val createLaoMetadata = laoMetadataMap.get(ActionType.CREATE).get
```

Comment:
these .get(...).get() (appears 3 times i believe) aren't really pretty ^^'. You can do `map(key)` instead of `map.get(key).get()` for the same result :)

Reply:
Yes thank you, it's been a while since I used maps in SCALA